charts/refinery/templates/NOTES.txt

```diff
@@ -1,13 +1,17 @@
-{{- if eq .Values.config.PeerManagement.Strategy "legacy" }}
-[INFO] We recommend setting "config.PeerManagement.Strategy" to "hash". See https://github.com/honeycombio/refinery/blob/main/RELEASE_NOTES.md#version-1200 for more details
-{{ end }}
-
 {{- if eq .Values.config.StressRelief.Mode "never" }}
 [INFO] We recommend setting "config.StressRelief.Mode" to "monitor". See https://github.com/honeycombio/refinery/blob/main/RELEASE_NOTES.md#version-1200 for more details
 {{ end }}
 
-{{- if eq .Values.config.SampleCacheConfig.Type "legacy" }}
-[INFO] We recommend setting "config.SampleCacheConfig.Type" to "cuckoo". See https://github.com/honeycombio/refinery/blob/main/RELEASE_NOTES.md#version-1200 for more details
+{{- if or ( (not (eq .Values.config.General.ConfigurationVersion 2)) (not (eq .Values.rules.RulesVersion 2)) ) }}
+[ERROR] Version 2.x of Refinery requires updated configuration and rules data. See https://github.com/honeycombio/refinery/blob/main/tools/convert/README.md for information on conversion.
```

Comment:
This will not actual fail the install/upgrade, it will only print out a line that says `[ERROR]`.  You need the `{{ fail }}` template command.  Example of this in action: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/e64cf3f0196bb8fd76eb16d90cc5edb0b5fb4d74/charts/opentelemetry-collector/templates/NOTES.txt#L14C1-L14C8